test/javascripts/integration/components/upcoming-events-list-test.gjs

```diff
@@ -0,0 +1,282 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import { render, waitFor } from "@ember/test-helpers";
+import UpcomingEventsList, {
+  DEFAULT_MONTH_FORMAT,
+  DEFAULT_DATE_FORMAT,
+  DEFAULT_TIME_FORMAT,
+} from "../../discourse/components/upcoming-events-list";
+import Service from "@ember/service";
+import pretender, { response } from "discourse/tests/helpers/create-pretender";
+import {
+  query,
+  queryAll,
+  fakeTime,
+} from "discourse/tests/helpers/qunit-helpers";
+import { pauseTest } from "@ember/test-helpers";
+import { hash } from "@ember/helper";
+
+class RouterStub extends Service {
+  currentRoute = { attributes: { category: { id: 1 } } };
+}
+
+const today = "2100-02-01T08:00:00";
+const tomorrowAllDay = "2100-02-02T00:00:00";
+const nextMonth = "2100-03-02T08:00:00";
+
+module("Integration | Component | upcoming-events-list", function (hooks) {
+  setupRenderingTest(hooks);
+
+  let clock;
+
+  hooks.beforeEach(function () {
+    this.owner.unregister("service:router");
+    this.owner.register("service:router", RouterStub);
+
+    // this.siteSettings.discourse_calendar_enabled = true;
+    // this.siteSettings.discourse_post_event_enabled = true;
```

Comment:
You probably don't need these two commented-out lines.

Reply:
Yup, leftover, will remove 👍